tensorflow/python/keras/layers/core.py

```diff
@@ -879,12 +879,15 @@ def compute_output_shape(self, input_shape):
       # is Graph mode because Lambda layers don't have side effects such as
       # `add_loss`.
       with context.eager_mode():
-        try:
-          return super(Lambda, self).compute_output_shape(input_shape)
-        except NotImplementedError:
-          raise NotImplementedError(
-              'We could not automatically infer the shape of the Lambda\'s '
-              'output. Please specify `output_shape` for this Lambda.')
+        input_shape = tf_utils.convert_shapes(input_shape, to_tuples=False)
+        def _make_placeholder_like(shape):
+          ph = K.placeholder(shape=shape, dtype=self.dtype)
+          ph._keras_mask = None
+          return ph
+
+        inputs = nest.map_structure(_make_placeholder_like, input_shape)
+        outputs = self.call(inputs)
+        return nest.map_structure(lambda t: t.shape, outputs)
```

Comment:
This is already what `super(Lambda, self).compute_output_shape(input_shape)` is supposed to do (`compute_output_shape` method of `Layer` base class). Why can't we leverage it, instead of duplicating the functionality?

Reply:
```_functional_construction_call``` under ```__call__``` causes RecusionError bug.

Reply:
I would recommend modifying the line `outputs = self(inputs, training=False)` to use `call` in `compute_output_shape` in `Layer`. That should be more generic, and it will avoid reimplementing functionality twice.

Reply:
Done